### PR TITLE
Company backend fixes / improvements (one fix one commit)

### PR DESCRIPTION
**Pull request by pigmej**
One fix one commit. I can rebase if needed though.

### CODE REVIEW

File: nim-mode.el [hunk 1]
```diff
@@ -981,7 +981,7 @@ The result is written into the buffer
 
 (cl-defstruct nim-epc section symkind qualifiedPath filePath forth line column doc)
 (defun nim-parse-epc (list)
-  (message "%S" list)
+  ;; (message "%S" list)
```
Comment (reactormonk) [inline nim-mode.el]: Just delete it.

File: company-nim.el [hunk 2]
```diff
@@ -66,21 +68,37 @@
...
+           thing))))
 
 
 (defun company-nim-annotation (cand)
-  (format " %s"
-          (get-text-property 0 :nim-type cand)
-          ;; (get-text-property 0 :nim-type cand)
-          ))
+  (let ((ann (get-text-property 0 :nim-type cand)))
+    (format " %s" (substring ann 0  (search "{" ann)))))
```
Comment (reactormonk) [inline company-nim.el]: You have a space too much after the `0`.

**Comment by reactormonk** [comment]
Oh, and merge against master. We break `ac-nim`, but meh.

**Comment by pigmej** [comment]
@reactormonk yeah, I will.

I improved today this backend.

Now we have:
* type abbrevs in completion
* proper documentation buffer (`C-h` on candidate)
* we can use company-quickhelp now to display help next to candidate
* proper location support (`C-w` on candidate works)

@reactormonk so if `ac-nim` will be broken on master, then I will remove `ac-nim.el`.

**Comment by reactormonk** [comment]
Nice work.

**Pull request merged**